dotnet/src/SemanticKernel.Abstractions/Contents/FunctionCallRequestContent.cs

```diff
@@ -0,0 +1,100 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+using System.Linq;
+using System.Text.Json.Serialization;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.SemanticKernel;
+
+/// <summary>
+/// Represents a function call requested by LLM.
+/// </summary>
+[Experimental("SKEXP0001")]
+public sealed class FunctionCallRequestContent : KernelContent
+{
+    /// <summary>
+    /// The function call ID.
+    /// </summary>
+    [JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingNull)]
+    public string? Id { get; }
+
+    /// <summary>
+    /// The plugin name.
+    /// </summary>
+    [JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingNull)]
+    public string? PluginName { get; }
+
+    /// <summary>
+    /// The function name.
+    /// </summary>
+    public string FunctionName { get; }
+
+    /// <summary>
+    /// The kernel arguments.
+    /// </summary>
+    [JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingNull)]
+    public KernelArguments? Arguments { get; }
+
+    /// <summary>
+    /// The exception that occurred while mapping original LLM function call request to the model class.
+    /// </summary>
+    [JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingNull)]
+    public Exception? Exception { get; init; }
+
+    /// <summary>
+    /// Creates a new instance of the <see cref="FunctionCallRequestContent"/> class.
+    /// </summary>
+    /// <param name="functionName">The function name.</param>
+    /// <param name="pluginName">The plugin name.</param>
+    /// <param name="id">The function call ID.</param>
+    /// <param name="arguments">The function original arguments.</param>
+    [JsonConstructor]
+    public FunctionCallRequestContent(string functionName, string? pluginName = null, string? id = null, KernelArguments? arguments = null)
+    {
+        Verify.NotNull(functionName);
+
+        this.FunctionName = functionName;
+        this.Id = id;
+        this.PluginName = pluginName;
+        this.Arguments = arguments;
+    }
+
+    /// <summary>
+    /// Invokes the function represented by the function call content type.
+    /// </summary>
+    /// <param name="kernel">The <see cref="Kernel"/> containing services, plugins, and other state for use throughout the operation.</param>
+    /// <param name="cancellationToken">The <see cref="CancellationToken"/> to monitor for cancellation requests. The default is <see cref="CancellationToken.None"/>.</param>
+    /// <returns>The result of the function's execution.</returns>
+    public async Task<FunctionCallResultContent> InvokeAsync(Kernel kernel, CancellationToken cancellationToken = default)
```

Comment:
Having function invocation logic inside `Content` class feels a little bit complicated to me. It would be better if we keep `Content` classes for data representation only. 

Is this something that is going to be used in Connectors in the future or there will be another abstraction/implementation for it? If I want to add new type of filter for function calling (i.e. `IFunctionCallFilter`) should I invoke it here? Just want to understand future plans for this method.

Reply:
Discussed offline, and agreed to find a proper place for the method when there's clear understanding of function-calling filters for manual function calling scenario.